firmware/baseband/proc_btlerx.cpp

```diff
@@ -26,245 +27,476 @@
+
+//     //tmp_bits = payload_bits((end-7) : end);
+//     //Hop = num2str( bi2de(tmp_bits(1:5), 'right-msb') );
+//     //SCA = num2str( bi2de(tmp_bits(6:end), 'right-msb') );
+//     payload_type_5->Hop = (payload_byte[33]&0x1F);
+//     payload_type_5->SCA = ((payload_byte[33]>>5)&0x07);
+//    }
+//     else
+//     {
+//    //TODO: Handle Unknown PDU.
+//      payload_type_R = (ADV_PDU_PAYLOAD_TYPE_R *)adv_pdu_payload;
+//      memcpy(payload_type_R->payload_byte, payload_byte, num_payload_byte);
+//     return(-1);
+//    }
+
+    return 0;
+}
+
 void BTLERxProcessor::execute(const buffer_c8_t& buffer) {
     if (!configured) return;
 
-    // FM demodulation
+    // Pulled this implementation from channel_stats_collector.c to time slice a specific packet's dB.
+    uint32_t max_squared = 0;
 
-    /*const auto decim_0_out = decim_0.execute(buffer, dst_buffer);
-        const auto channel = decim_1.execute(decim_0_out, dst_buffer);
+    void* src_p = buffer.p;
 
-        feed_channel_stats(channel);
+    while (src_p < &buffer.p[buffer.count]) {
+        const uint32_t sample = *__SIMD32(src_p)++;
+        const uint32_t mag_sq = __SMUAD(sample, sample);
+        if (mag_sq > max_squared) {
+            max_squared = mag_sq;
+        }
+    }
 
-        auto audio_oversampled = demod.execute(channel, work_audio_buffer);*/
+    const float max_squared_f = max_squared;
+    const int32_t max_dB = mag2_to_dbv_norm(max_squared_f * (1.0f / (32768.0f * 32768.0f)));
 
-    const auto decim_0_out = decim_0.execute(buffer, dst_buffer);
-    feed_channel_stats(decim_0_out);
+    decim_0.execute(buffer, dst_buffer);
+    feed_channel_stats(dst_buffer);
 
-    auto audio_oversampled = demod.execute(decim_0_out, work_audio_buffer);
+    const buffer_c8_t iq_buffer{
+        buffer.p,
+        buffer.count,
+        baseband_fs};
 
-    /*std::fill(spectrum.begin(), spectrum.end(), 0);
-        for(size_t i=0; i<spectrum.size(); i++) {
-                spectrum[i] += buffer.p[i];
-        }
-        const buffer_c16_t buffer_c16 {spectrum.data(),spectrum.size(),buffer.sampling_rate};
-        feed_channel_stats(buffer_c16);
+    // process++;
 
-        auto audio_oversampled = demod.execute(buffer_c16, work_audio_buffer);*/
-    // Audio signal processing
-    for (size_t c = 0; c < audio_oversampled.count; c++) {
-        /*const int32_t sample_int = audio_oversampled.p[c] * 32768.0f;
-                int32_t current_sample = __SSAT(sample_int, 16);
-                current_sample /= 128;*/
+    // if ((process % 50) != 0) return;
 
-        int32_t current_sample = audio_oversampled.p[c];  // if I directly use this, some results can pass crc but not correct.
-        rb_head++;
-        rb_head = (rb_head) % RB_SIZE;
+    // 4Mhz 2048 samples
 
-        rb_buf[rb_head] = current_sample;
+    //--------------Variable Defines---------------------------------//
 
-        skipSamples = skipSamples - 1;
+    int i, sp, j = 0;
```

Comment:
nit: as mentioned, pre-decl isn't necessary in C++ so you don't _have_ to do this (but it's fine).

Reply:
C dev habit lol.